spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -84,7 +85,7 @@ static void validateConfigParameters(Schema inputSchema, @Nullable String featur
       throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
                                            "Please specify fields for one.");
     }
-    Map<String, Integer> fields = new HashMap<>();
+    Map<String, Integer> fields = new LinkedHashMap<>();
```

Comment:
The test checks the schema for field types and throws error if the type is not in : int, long, float or double.
In the input schema the "tailNum" is the first field that does not satisfy the condition, and therefore is expected to throw error on this field.
However, since HashMap does not maintain the insertion order the keySet received from getFeatureList() was in random order even if "inputSchemaFields"  had the correct order of elements and caused the test to fail at field other than "tailNum".
Since, LinkedList maintains the insertion order the keySet retrieved from the getFeatureList() was in the same order as is present in "inputSchemaFields" and hence the test case passes as it now throws error on the first non-numeric field in the schema i.e. "tailNum" as is expected by the test.
Please let me know if this answers the question.